OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -150,8 +150,38 @@ protected override void OnFirstRun(Actor self)
 			}
 		}
 
+		public override bool Cancel(Actor self, bool keepQueue = false)
+		{
+			// Let child activities do what they want so that units don't end up in strange state.
+			// Just clear the path so that child activity doesn't run for too long.
+			// Although we always return false here, Move with nowhere to move will eventually
+			// proceed to the NextActivity when the actor is in good position.
+			if (ChildActivity == null)
+				return base.Cancel(self, keepQueue);
+
+			if (path != null)
+				path.Clear();
+
+			if (keepQueue == false && NextActivity != null)
+				NextActivity.Cancel(self, keepQueue);
+
+			// Could be turn activity, which should be canceled.
+			// To prevent Move first/second half activities from being interrupted,
+			// we set IsInterruptible to false in their constructor.
+			return ChildActivity.Cancel(self, false);
+		}
+
 		public override Activity Tick(Actor self)
 		{
+			// ChildActivity is the top priority, unlike other activities.
+			// Even if this activity is canceled, we must let the child be run so that units
+			// will not end up in an odd place.
+			if (ChildActivity != null)
+			{
+				ActivityUtils.RunActivity(self, ChildActivity);
```

Comment:
I strongly think it wrong because that would prevent an activity from running multiple children activities properly. They do run, as you might see in my simultaneous reload + resupply code but ChildActivity pointer is invalid.

https://github.com/OpenRA/OpenRA/pull/13733/files/a16b9b52c295ea5c24ed2bba71a159275bd260cc#diff-268b61855646bbc4ed78d90265240289R43